src/modules/formBuilder/formBuilderUtil.ts

```diff
@@ -214,3 +214,79 @@ export const insertItemToDefinition = ({
     newParentArray.splice(insertAtIndex, 0, item);
   }
 };
+
+/**
+ * Validates that the attempted deletion action is allowed. It is allowed if the item
+ * the user is trying to delete doesn't have any other items that refer to it.
+ * If dependents exist, then the operation is not allowed and the function returns a user-facing error string.
+ *
+ * @param toDelete the linkId of the item the user is trying to delete
+ * @param definition the current form definition
+ */
+export const validateDeletion = ({
+  toDelete,
+  definition,
+}: {
+  toDelete: string;
+  definition: FormDefinitionJson;
+}): { success: boolean; errors?: Set<string> } => {
+  // Mapping of linkId of the dependent question => an array of user-facing strings
+  // representing where the problematic dependency is located on that item, e.g. autofill or enableWhen
+  const errors = new Set<string>();
+
+  function checkDependence(
+    toDelete: string,
+    toCheck: FormItem,
+    parentPath: string
```

Comment:
Tiny thing, but I would rename this. (I know I have a separate comment that might make this irrelevant). We use "parentPath"/"itemPath" elsewhere in the code to describe paths like `item.0.item.5`, which I think is pretty intuitive. Here it's describing `"label > label > label"`, I think, which feels less intuitive. Maybe call this `parentLabelPrefix` or `parentLabels` (as an array) or simply `labelPrefix`.

In general imo this helps with code readability if we have consistent meanings for words like "path" and "linkId" etc. Not always possible to be 100% consistent but we can try :)

Reply:
Yes good point, thanks for this! (didn't end up making any changes here because, as you said, your other comment meant this code got deleted)